.github/workflows/ci_test-full.yml

```diff
@@ -108,58 +115,77 @@ jobs:
         pip list
       shell: bash
 
-    - name: Reinstall Horovod if necessary
-      if: runner.os != 'windows'
-      env:
-        HOROVOD_BUILD_ARCH_FLAGS: "-mfma"
-        HOROVOD_WITHOUT_MXNET: 1
-        HOROVOD_WITHOUT_TENSORFLOW: 1
-      run: |
-        HOROVOD_BUILT=$(python -c "import horovod.torch; horovod.torch.nccl_built(); print('SUCCESS')" || true)
-        if [[ $HOROVOD_BUILT != "SUCCESS" ]]; then
-          pip uninstall -y horovod
-          echo $(grep "horovod" requirements/extra.txt) > requirements/horovod.txt
-          pip install --no-cache-dir -r requirements/horovod.txt
-        fi
-        horovodrun --check-build
-      shell: bash
+    # - name: Reinstall Horovod if necessary
```

Comment:
This is not a good way to disable horovod, because it's already available in the docker image used by the job.

For this to work, you'd need to explicitly uninstall, not to stop installing.

I'd try hard-coding `_HOROVOD_AVAILABLE = False` so the tests get all skipped.

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/blob/cdc1bb1a6874ce0106766f8a862715b8b6b90564/.github/workflows/ci_test-full.yml#L64

Is it really? I thought excluding horovod installation above should skip all tests needing horovod as the docker image used by this job is normal ubuntu.

I'll try hard-coding it and uninstall it as you suggest anyway now :]

Reply:
NVM you are entirely right :)